pkg/gadgets/trace/bind/tracer/tracer_test.go

```diff
@@ -0,0 +1,649 @@
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+			},
+			generateEvent: bindSocketFn("::", unix.AF_INET6, unix.SOCK_STREAM, 0),
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, port uint16, events []types.Event) {
+				if len(events) != 1 {
+					t.Fatalf("Wrong number of events received %d, expected 1", len(events))
+				}
+
+				if events[0].Addr != "::" {
+					t.Fatalf("Event has wrong addr: %s, expected: %s",
+						events[0].Addr, "::")
+				}
+			},
+		},
+		"udp": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+			},
+			generateEvent: bindSocketFn("127.0.0.1", unix.AF_INET, unix.SOCK_DGRAM, 0),
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, port uint16, events []types.Event) {
+				if len(events) != 1 {
+					t.Fatalf("Wrong number of events received %d, expected 1", len(events))
+				}
+
+				if events[0].Protocol != "UDP" {
+					t.Fatalf("Event has wrong protocol: %s, expected: %s",
+						events[0].Protocol, "UDP")
+				}
+			},
+		},
+		"tcp": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+			},
+			generateEvent: bindSocketFn("127.0.0.1", unix.AF_INET, unix.SOCK_STREAM, 0),
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, port uint16, events []types.Event) {
+				if len(events) != 1 {
+					t.Fatalf("Wrong number of events received %d, expected 1", len(events))
+				}
+
+				if events[0].Protocol != "TCP" {
+					t.Fatalf("Event has wrong protocol: %s, expected: %s",
+						events[0].Protocol, "TCP")
+				}
+			},
+		},
+		"interface": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+			},
+			generateEvent: func() (uint16, error) {
+				opts := []sockOpt{
+					{
+						level: unix.SOL_SOCKET,
+						opt:   unix.SO_BINDTOIFINDEX,
+						value: 1, // "lo" iface
+					},
+				}
+
+				return bindSocketWithOpts("127.0.0.1", unix.AF_INET, unix.SOCK_STREAM, 0, opts)
+			},
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, port uint16, events []types.Event) {
+				if len(events) != 1 {
+					t.Fatalf("Wrong number of events received %d, expected 1", len(events))
+				}
+
+				if events[0].Interface != "lo" {
+					t.Fatalf("Event has wrong interface: %s, expected: %s",
+						events[0].Interface, "lo")
+				}
+			},
+		},
+		"pid_filter_match": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					TargetPid: int32(info.Pid),
+					// It's difficult to only test
+					// the PID filter since other
+					// test events are generated
+					// from the same PID on this
+					// test suite
```

Comment:
You're right. Unfortunately from an implementation point of view this will be difficult as we'll need to fork the process generating the events and create an RPC to send events from one process to another. Maybe for now one option is to continue testing the pid and mountns id filter together.